src/compiler/checker.ts

```diff
@@ -27526,7 +27526,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 location = location.parent;
             }
             if (isExpressionNode(location) && (!isAssignmentTarget(location) || isWriteAccess(location))) {
-                const type = getTypeOfExpression(location as Expression);
+                const type = location.flags & NodeFlags.OptionalChain ?
+                    checkPropertyAccessChain(location as PropertyAccessChain, CheckMode.Normal, /*wasOptional*/ false) :

```

Comment:
Note that this might not contain some guard rails related to control flow analysis that are present in `getTypeOfExpression`. I'm not sure if they are relevant for this case though.

Reply:
Wouldn't this work as well?

```ts
const type = removeOptionalTypeMarker(getTypeOfExpression(location as Expression));
```

For an expression like `a?.b.c`, hovering over `c` would still show `C | undefined` (since its the end of the chain and the optional type marker is replaced with `undefined`), but hovering over `b` would just show `B`.

The upside is that `getTypeOfExpression` caches, so you avoid recomputing a type that's already cached.

Reply:
There's no need to make this conditional on `location.flags & NodeFlags.OptionalChain`. The marker type will only be present for optional chains.

Reply:
cool, removed this redundant check 👍